numpy/core/src/multiarray/ctors.c

```diff
@@ -713,14 +713,20 @@ discover_itemsize(PyObject *s, int nd, int *itemsize, int string_type)
     return 0;
 }
 
+typedef enum {
+    DISCOVERED_OK = 0,
+    DISCOVERED_RAGGED = 1,
+    DISCOVERED_OBJECT = 2
+} discovered_t;
```

Comment:
Is there precedent for a `_t` suffixin numpy? In principle, this is reserved by POSIX.

Reply:
For enums, `mem_overlap.h` has `mem_overlap_t`. There are many sructs that use `_t`. It is easy enough to change, what would you prefer?

Reply:
If we already do it, then we've already crossed the posix-compliant line, so there's no point worrying here - fine as is, we could consider fixing them all if anyone complains.